plugin.rb

```diff
@@ -30,19 +36,98 @@ class Engine < ::Rails::Engine
   end
 
   [
+    '../app/controllers/salesforce/admin_controller.rb',
+    '../app/controllers/salesforce/cases_controller.rb',
     '../app/controllers/salesforce/persons_controller.rb',
+    '../app/models/salesforce/case.rb',
     '../app/models/salesforce/person.rb',
+    '../app/serializers/concerns/case_mixin.rb',
+    '../app/serializers/case_serializer.rb',
     '../lib/salesforce/api.rb'
   ].each { |path| load File.expand_path(path, __FILE__) }
 
   Salesforce::Engine.routes.draw do
     post "/persons/create" => "persons#create"
+    post "/cases/sync" => "cases#sync"
   end
 
+  add_admin_route 'salesforce.title', 'salesforce'
+
   Discourse::Application.routes.append do
     mount ::Salesforce::Engine, at: "salesforce"
+    get '/admin/plugins/salesforce' => 'admin/plugins#index', constraints: AdminConstraint.new
+    get '/admin/plugins/salesforce/index' => 'salesforce/admin#index', constraints: AdminConstraint.new
   end
 
+  allow_staff_user_custom_field(::Salesforce::Person::CONTACT_ID_FIELD)
+  allow_staff_user_custom_field(::Salesforce::Person::LEAD_ID_FIELD)
+
+  reloadable_patch do |plugin|
+    require_dependency 'user'
+    require_dependency 'topic'
+
+    class ::User
+      def salesforce_contact_id
+        custom_fields[::Salesforce::Person::CONTACT_ID_FIELD]
+      end
+
+      def create_salesforce_contact
+        ::Salesforce::Person.create!("contact", self)
+      end
+    end
+
+    class ::Topic
+      def has_salesforce_case
+        custom_fields["has_salesforce_case"] ? true : false
+      end
+
+      def salesforce_case
+        return unless has_salesforce_case
+        ::Salesforce::Case.find_by(topic_id: id)
+      end
+    end
+
+    require_dependency 'topic_list_item_serializer'
+    require_dependency 'search_topic_list_item_serializer'
+    require_dependency 'suggested_topic_serializer'
+    require_dependency 'user_summary_serializer'
```

Comment:
I'm curious... are these required? I *think* these dependencies should all be available automatically, because we're inside an after_initialize block 🤔  Not 100% sure though.

(same goes for the `require_dependency 'user', 'topic' above)